.github/workflows/build.yml

```diff
@@ -15,30 +15,38 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.8', '3.9', '3.10', '3.11']
 
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-        pip install -e .
 
+    - name: Set up a virtual environment for Python ${{ matrix.python-version }}
```

Comment:
As the gh actions work in isolated environments already I dont think we need to create a venv here. In CICD time is important so I would not want to waste even a couple of seconds for creating a venv for nothing. This machine will be garbage after we use (probably GH actions use containers or similar)

Reply:
I can generate req.txt file with devs and install and do tests. So we can skip poetry/venv section

Reply:
Oh does poetry only work with venv?

Reply:
> Oh does poetry only work with venv?

Poetry logic is work with "venv" so yes. If you just fire up "poetry install" it will create venv but you can turn off by using 

`poetry config virtualenvs.create false ` and it should be install into current env, well now come to think of I think I can disable venv and do install and etc. :)

Reply:
Ok but in this one you create the venv with `virtualenv venv` not with poetry. so you could just install poetry in the machine not in a venv and run `poetry install` there. Is it feasible?

Reply:
Because if you use a venv at each step you should be using a `source venv/bin/activate` which seems an overkill.

Reply:
I removed venv so all done.